libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_stress_test.c

```diff
@@ -58,78 +62,123 @@ TEST_GROUP_RUNNER( Full_BLE_Stress_Test )
     /* Initializations that need to be done once for all the tests. */
     RUN_TEST_CASE( Full_BLE, BLE_Setup );
 
-    for( uint32_t init_loop = 0; init_loop < INIT_DEINIT_NUMBER_STRESS_TEST; init_loop++ )
+    for( uint32_t init_loop = 0; init_loop < STRESS_TEST_INIT_REPETITION; init_loop++ )
     {
-        RUN_TEST_CASE( Full_BLE_Stress_Test, BLE_Stack_Init );
+        RUN_TEST_CASE( Full_BLE_Stress_Test, Initialize_common_GAP );
 
-        for( uint32_t enable_loop = 0; enable_loop < ENABLE_DISABLE_NUMBER_STRESS_TEST; enable_loop++ )
+        for( uint32_t enable_loop = 0; enable_loop < STRESS_TEST_ENABLE_REPETITION; enable_loop++ )
         {
             /*TODO: add some randomness to catch timing issues */
+            RUN_TEST_CASE( Full_BLE_Stress_Test, BLE_Init_GAP_GATT );
+            RUN_TEST_CASE( Full_BLE, BLE_Advertising_SetProperties );       /*@TOTO, incomplete */
+            RUN_TEST_CASE( Full_BLE, BLE_Connection_RemoveAllBonds );
+            RUN_TEST_CASE( Full_BLE, BLE_Advertising_SetAvertisementData ); /*@TOTO, incomplete */
 
-            RUN_TEST_CASE( Full_BLE_Stress_Test, BLE_Stack_Enable );
-
-            RUN_TEST_CASE( Full_BLE, BLE_Initialize_BLE_GAP );
-            RUN_TEST_CASE( Full_BLE, BLE_Initialize_BLE_GATT );
+            for( uint16_t loop = 0; loop < STRESS_TEST_MODULE_REPETITION; loop++ )
+            {
+                RUN_TEST_CASE( Full_BLE_Stress_Test, BLE_Connection_ReConnect );
+            }
 
             RUN_TEST_CASE( Full_BLE_Stress_Test, BLE_Service_Create );
-            RUN_TEST_CASE( Full_BLE_Stress_Test, BLE_Service_Restart );
-            RUN_TEST_CASE( Full_BLE_Stress_Test, BLE_Service_Delete );
 
-            RUN_TEST_CASE( Full_BLE, BLE_Advertising_SetProperties );       /*@TOTO, incomplete */
-            RUN_TEST_CASE( Full_BLE, BLE_Connection_RemoveAllBonds );
-            RUN_TEST_CASE( Full_BLE, BLE_Advertising_SetAvertisementData ); /*@TOTO, incomplete */
+            for( uint16_t loop = 0; loop < STRESS_TEST_MODULE_REPETITION; loop++ )
+            {
+                RUN_TEST_CASE( Full_BLE_Stress_Test, BLE_Service_Restart );
+            }
 
-            RUN_TEST_CASE( Full_BLE_Stress_Test, BLE_Connection_ReConnect );
+            RUN_TEST_CASE( Full_BLE_Stress_Test, BLE_Start_Connection );
 
-            RUN_TEST_CASE( Full_BLE_Stress_Test, BLE_Teardown );
+            for( uint16_t loop = 0; loop < STRESS_TEST_MODULE_REPETITION; loop++ )
```

Comment:
Will the test loop 100 times even if one test case fails ? Should we fail fast in that case?

Reply:
The limitation of the framework is that, if it fails during the connect/disconnect or notification/indication, since RPi hangs on mainloop.run(), it could fail the tests that follow.